### PR TITLE
Cites trade eu query

### PR DESCRIPTION
**Pull request by lucacug**
This includes a refactor to the EU query present in this #894 (that can be superseded by this if you happy) to include all the edge cases (a country that left EU might rejoin, EU + country selection (EU+ Austria)) and fix the tests.

I left a few comments on the code to facilitate the review, some of which we would need to remove before merging.

Any issues please let me know, happy to do a PairProgramming session code review if easier

**Note**: Translations for EU disclaimer still need adding when ready.

### CODE REVIEW

File: app/models/trade/filter.rb [hunk 1]
```diff
@@ -79,13 +79,38 @@ def initialize_query
...
+      @query = @query.where(:importer_id => importers_ids)
     end
 
     unless @exporters_ids.empty?
-      @query = @query.where(:exporter_id => @exporters_ids)
+      exporters_ids = sanitize_exporter_ids(@exporters_ids)
+      @query = @query.where(:exporter_id => exporters_ids)
     end
+    #
+    # unless @exporters_ids.empty?
```
Comment (roshan-mathewtech) [inline app/models/trade/filter.rb:103]: Same here

**Review by roshan-mathewtech — COMMENTED** [review]
Looks good. Just added a small comment on filter.rb
